subscription/views.py

```diff
@@ -1,106 +1,131 @@
 import stripe
 
 from django.conf import settings
-from django.contrib.auth.decorators import login_required
-from django.contrib.auth.models import User
-from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.http import HttpResponse, JsonResponse
-from django.shortcuts import redirect, render
-from django.utils.decorators import method_decorator
+from django.shortcuts import render
 from django.views.generic import TemplateView, View
 
 from subscription.models import Customer, Product
 
 stripe.api_key = settings.STRIPE_SECRET_KEY
 
 
-class StripeConfiguration(View):
-    @method_decorator(login_required)
+class StripeConfiguration(LoginRequiredMixin, View):
+    """
+    Get the sesssion id based on Stripe primary key.
+    """
+
     def get(self, request):
-        stripe_config = {'publicKey': settings.STRIPE_PUBLISHABLE_KEY}
+        stripe_config = {"publicKey": settings.STRIPE_PUBLISHABLE_KEY}
         context = JsonResponse(stripe_config, safe=False)
         return HttpResponse(context)
 
 
-class CreateCheckoutSessionView(View):
-    @method_decorator(login_required)
+class CreateCheckoutSessionView(LoginRequiredMixin, View):
+    """
+    Create the checkout session for the subscription.
+    """
+
     def get(self, request):
-        domain_url = 'http://localhost:8000/'
+        domain_url = "http://localhost:8000/"
         try:
             checkout_session = stripe.checkout.Session.create(
-                client_reference_id=request.user.id if request.user.is_authenticated else None,
-                success_url=domain_url +
-                'success?session_id={CHECKOUT_SESSION_ID}',
-                cancel_url=domain_url + 'cancel/',
-                payment_method_types=['card'],
-                mode='subscription',
-                subscription_data={'trial_period_days': 7},
-                line_items=[
-                    {
-                        'price': settings.STRIPE_PRICE_ID,
-                        'quantity': 1,
-                    }
-                ],
+                client_reference_id=request.user.id,
+                success_url=domain_url + "success?session_id={CHECKOUT_SESSION_ID}",
+                cancel_url=domain_url + "cancel/",
+                payment_method_types=["card"],
+                mode="subscription",
+                subscription_data={"trial_period_days": 7},
+                line_items=[{"price": settings.STRIPE_PRICE_ID, "quantity": 1}],
             )
-            context = JsonResponse({'sessionId': checkout_session['id']})
+            context = JsonResponse({"sessionId": checkout_session["id"]})
 
         except Exception as e:
-            context = JsonResponse({'error': str(e)})
+            context = JsonResponse({"error": str(e)})
```

Comment:
Use Rest Framework response or you can also change this to an API's also as the response is JSON!